base/libc.jl

```diff
@@ -402,6 +402,31 @@ Interface to the C `srand(seed)` function.
 """
 srand(seed=floor(Int, time()) % Cuint) = ccall(:srand, Cvoid, (Cuint,), seed)
 
+"""
+    mkfifo(path::AbstractString, [mode::Integer]) -> path
+
+Make a FIFO special file (a named pipe) at `path`.  Return `path` as-is on success.
+
+`mkfifo` is supported only in Unix platforms.
+
+!!! compat "Julia 1.8"
+    `mkfifo` requires at least Julia 1.8.
+"""
+function mkfifo(
+    path::AbstractString,
+    mode::Integer = Base.S_IRUSR | Base.S_IWUSR | Base.S_IRGRP | Base.S_IWGRP |
+                    Base.S_IROTH | Base.S_IWOTH,
+)
+    @static if Sys.isunix()
+        # Default `mode` is compatible with `mkfifo` CLI in coreutils.
+        ret = ccall(:mkfifo, Cint, (Cstring, Base.Cmode_t), path, mode)
+        systemerror("mkfifo", ret == -1)
+        return path
+    else
+        systemerror("mkfifo", ENOTSUP)
```

Comment:
BTW, I noticed that `ENOTSUP` doesn't exist in my local build. I guess it's OK as long as this is defined in Windows, though.

Reply:
Looks like Linux defines EOPNOTSUP instead, and our regex isn't smart enough to pick up the value for ENOTSUP from that. But CI seems to be okay with this.

Reply:
It seems Windows prints `ENOTSUP` as "Unknown error":

>     Error in testset file:
>     Test Failed at C:\buildbot\worker-tabularasa\tester_win64\build\share\julia\test\file.jl:1635
>       Expression: Libc.mkfifo(joinpath(pwd(), "dummy_path"))
>         Expected: "mkfifo: Operation not supported"
>          Message: "SystemError: mkfifo: Unknown error"
>
> --- https://build.julialang.org/#/builders/65/builds/5360

So, I suggest simply doing:

https://github.com/JuliaLang/julia/blob/02961cc2b25031cb83628342f4471adbc55c9cf1/base/libc.jl#L426-L428

Is it OK?